ed25519/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+extern crate ed25519_dalek;
+extern crate ewasm_api;
+extern crate sha2;
+
+mod verify;
+
+#[cfg(not(test))]
+#[no_mangle]
+pub extern "C" fn main() {
+    let length = ewasm_api::calldata_size();
+
+    ewasm_api::consume_gas(2000);
```

Comment:
Maybe move this gas usage statement to the top of the function body, for cleanliness?

Reply:
yeah

Reply:
Done.